pkg/provision/storage/perWorkspaceStorage.go

```diff
@@ -213,6 +213,7 @@ func syncPerWorkspacePVC(workspace *common.DevWorkspaceWithConfig, clusterAPI sy
 	if err != nil {
 		return nil, err
 	}
+	pvc.ObjectMeta.Labels[constants.DevWorkspaceIDLabel] = workspace.Status.DevWorkspaceId
```

Comment:
I believe the tests are failing, since `pvc.ObjectMeta.Labels` could be nil (and so could `pvc.ObjectMeta` it seems, since it's marked as optional). The labels  (and maybe ObjectMeta) should be initialized before indexing into it.